python/tests/avro/test_reader.py

```diff
@@ -0,0 +1,162 @@
+        "avro.codec": "deflate",
+        "format-version": "1",
+        "partition-spec-id": "0",
+        "iceberg.schema": '{"type":"struct","schema-id":0,"fields":[{"id":0,"name":"status","required":true,"type":"int"},{"id":1,"name":"snapshot_id","required":false,"type":"long"},{"id":2,"name":"data_file","required":true,"type":{"type":"struct","fields":[{"id":100,"name":"file_path","required":true,"type":"string","doc":"Location URI with FS scheme"},{"id":101,"name":"file_format","required":true,"type":"string","doc":"File format name: avro, orc, or parquet"},{"id":102,"name":"partition","required":true,"type":{"type":"struct","fields":[{"id":1000,"name":"VendorID","required":false,"type":"int"}]}},{"id":103,"name":"record_count","required":true,"type":"long","doc":"Number of records in the file"},{"id":104,"name":"file_size_in_bytes","required":true,"type":"long","doc":"Total file size in bytes"},{"id":105,"name":"block_size_in_bytes","required":true,"type":"long"},{"id":108,"name":"column_sizes","required":false,"type":{"type":"map","key-id":117,"key":"int","value-id":118,"value":"long","value-required":true},"doc":"Map of column id to total size on disk"},{"id":109,"name":"value_counts","required":false,"type":{"type":"map","key-id":119,"key":"int","value-id":120,"value":"long","value-required":true},"doc":"Map of column id to total count, including null and NaN"},{"id":110,"name":"null_value_counts","required":false,"type":{"type":"map","key-id":121,"key":"int","value-id":122,"value":"long","value-required":true},"doc":"Map of column id to null value count"},{"id":137,"name":"nan_value_counts","required":false,"type":{"type":"map","key-id":138,"key":"int","value-id":139,"value":"long","value-required":true},"doc":"Map of column id to number of NaN values in the column"},{"id":125,"name":"lower_bounds","required":false,"type":{"type":"map","key-id":126,"key":"int","value-id":127,"value":"binary","value-required":true},"doc":"Map of column id to lower bound"},{"id":128,"name":"upper_bounds","required":false,"type":{"type":"map","key-id":129,"key":"int","value-id":130,"value":"binary","value-required":true},"doc":"Map of column id to upper bound"},{"id":131,"name":"key_metadata","required":false,"type":"binary","doc":"Encryption key metadata blob"},{"id":132,"name":"split_offsets","required":false,"type":{"type":"list","element-id":133,"element":"long","element-required":true},"doc":"Splittable offsets"},{"id":140,"name":"sort_order_id","required":false,"type":"int","doc":"Sort order ID"}]}}]}',
+        "partition-spec": '[{"name":"VendorID","transform":"identity","source-id":1,"field-id":1000}]',
+    }
+    assert header.sync == b"B\x9e\xb0\xc3\x04,vG\xd1O\x8e\xf5\x9c\xdd\x17\x8d"
```

Comment:
Should we also test `header.get_schema()`?

```suggestion
    assert header.sync == b"B\x9e\xb0\xc3\x04,vG\xd1O\x8e\xf5\x9c\xdd\x17\x8d"
    assert header.get_schema() == Schema(
        NestedField(field_id=0, name="status", field_type=IntegerType(), is_optional=False),
        NestedField(
            field_id=1, name="snapshot_id", field_type=LongType(), is_optional=True
        ),
        NestedField(
            field_id=2,
            name="data_file",
            field_type=StructType(
                fields=(
                    NestedField(
                        field_id=100,
                        name="file_path",
                        field_type=StringType(),
                        is_optional=False,
                    ),
                    NestedField(
                        field_id=101,
                        name="file_format",
                        field_type=StringType(),
                        is_optional=False,
                    ),
                    NestedField(
                        field_id=102,
                        name="partition",
                        field_type=StructType(
                            fields=(
                                NestedField(
                                    field_id=1000,
                                    name="VendorID",
                                    field_type=IntegerType(),
                                    is_optional=True,
                                ),
                            )
                        ),
                        is_optional=False,
                    ),
                    NestedField(
                        field_id=103,
                        name="record_count",
                        field_type=LongType(),
                        is_optional=False,
                    ),
                    NestedField(
                        field_id=104,
                        name="file_size_in_bytes",
                        field_type=LongType(),
                        is_optional=False,
                    ),
                    NestedField(
                        field_id=105,
                        name="block_size_in_bytes",
                        field_type=LongType(),
                        is_optional=False,
                    ),
                    NestedField(
                        field_id=108,
                        name="column_sizes",
                        field_type=MapType(
                            key_id=117,
                            key_type=IntegerType(),
                            value_id=118,
                            value_type=LongType(),
                            value_is_optional=False,
                        ),
                        is_optional=True,
                    ),
                    NestedField(
                        field_id=109,
                        name="value_counts",
                        field_type=MapType(
                            key_id=119,
                            key_type=IntegerType(),
                            value_id=120,
                            value_type=LongType(),
                            value_is_optional=False,
                        ),
                        is_optional=True,
                    ),
                    NestedField(
                        field_id=110,
                        name="null_value_counts",
                        field_type=MapType(
                            key_id=121,
                            key_type=IntegerType(),
                            value_id=122,
                            value_type=LongType(),
                            value_is_optional=False,
                        ),
                        is_optional=True,
                    ),
                    NestedField(
                        field_id=137,
                        name="nan_value_counts",
                        field_type=MapType(
                            key_id=138,
                            key_type=IntegerType(),
                            value_id=139,
                            value_type=LongType(),
                            value_is_optional=False,
                        ),
                        is_optional=True,
                    ),
                    NestedField(
                        field_id=125,
                        name="lower_bounds",
                        field_type=MapType(
                            key_id=126,
                            key_type=IntegerType(),
                            value_id=127,
                            value_type=BinaryType(),
                            value_is_optional=False,
                        ),
                        is_optional=True,
                    ),
                    NestedField(
                        field_id=128,
                        name="upper_bounds",
                        field_type=MapType(
                            key_id=129,
                            key_type=IntegerType(),
                            value_id=130,
                            value_type=BinaryType(),
                            value_is_optional=False,
                        ),
                        is_optional=True,
                    ),
                    NestedField(
                        field_id=131,
                        name="key_metadata",
                        field_type=BinaryType(),
                        is_optional=True,
                    ),
                    NestedField(
                        field_id=132,
                        name="split_offsets",
                        field_type=ListType(
                            element_id=133,
                            element_type=LongType(),
                            element_is_optional=False,
                        ),
                        is_optional=True,
                    ),
                    NestedField(
                        field_id=140,
                        name="sort_order_id",
                        field_type=IntegerType(),
                        is_optional=True,
                    ),
                )
            ),
            is_optional=False,
        ),
        schema_id=1,
        identifier_field_ids=[],
    )
```

Reply:
I've added it to the top of the file to not bloat the conftest.py